pennylane/_qubit_device.py

```diff
@@ -380,18 +380,47 @@ def execute_new(self, circuit, **kwargs):
         # apply all circuit operations
         self.apply(circuit.operations, rotations=circuit.diagonalizing_gates, **kwargs)
 
-        results = self.statistics(circuit.observables)
+        # generate computational basis samples
+        if self.shots is not None or circuit.is_sampled:
+            self._samples = self.generate_samples()
 
-        if len(circuit.measurements) == 1:
-            if circuit.measurements[0].return_type is qml.measurements.State:
-                # State: assumed to only be allowed if it's the only measurement
-                results = self._asarray(results[0], dtype=self.C_DTYPE)
-            else:
-                # All other measurements are real
-                results = self._asarray(results[0], dtype=self.R_DTYPE)
+        # compute the required statistics
+        if not self.analytic and self._shot_vector is not None:
```

Comment:
```suggestion
        if self._shot_vector is not None:
```
I think it might be sufficient, because you cannot be analytic with a shot vector. What is the different between `analytic` and finite `shots`?

Reply:
This is a remnant from the past, also wondering about this :thinking: `self.analytic` should be just a convenience that we used to have in PennyLane, but not anymore.